.github/workflows/test.yml

```diff
@@ -0,0 +1,59 @@
+name: Python tests
+
+on:
+    push:
+        branches:
+        - '**'
+    pull_request:
+        branches:
+        - main
+        types: [opened, reopened]
+
+jobs:
+  black:
+    name: Check formatting with black
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.9"]
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install black
```

Comment:
This will always install the latest version of black at the moment. Which is going to mean over-time this action will start failing. We should set it to a specific version, we currently use `black==21.7b0` in gitlab